src/rgw/rgw_rest_s3.cc

```diff
@@ -3957,11 +3956,13 @@ int RGWHandler_REST_S3::init(RGWRados *store, struct req_state *s,
   ret = rgw_validate_tenant_name(s->bucket_tenant);
   if (ret)
     return ret;
-  bool relaxed_names = s->cct->_conf->rgw_relaxed_s3_bucket_names;
   if (!s->bucket_name.empty()) {
-    ret = valid_s3_bucket_name(s->bucket_name, relaxed_names);
-    if (ret)
-      return ret;
+    if (s->op_type == RGW_OP_CREATE_BUCKET) {
+      bool relaxed_names = s->cct->_conf->rgw_relaxed_s3_bucket_names;
+      ret = valid_s3_bucket_name(s->bucket_name, relaxed_names);
```

Comment:
i don't think s->op_type is initialized yet when this is called. `RGWCreateBucket_ObjStore_S3::get_params()` is probably the best place to do this validation. it shouldn't need to be duplicated in `RGWHandler_REST_S3::postauth_init()`

Reply:
Done.